techniques/ios/MASTG-TECH-0066.md

```diff
@@ -6,3 +6,109 @@ platform: ios
+
+```bash
+afl~SecKeyCreateRandomKey
+```
+
+Example: In @MASTG-DEMO-0011, `SecKeyCreateRandomKey` is identified as a function of interest.
+
+**When Not to Use `afl`**:
+
+In some cases, functions may not appear in the list generated by `afl`. For example, `canEvaluatePolicy` is a dynamically resolved function and may not be listed.
+
+```bash
+afl~canEvaluatePolicy
+< no results >
+```
+
+Instead, use [flags](https://book.rada.re/commandline/flags.html) (`f`), string searches with `/`, `/z`, `iz~` and then use [cross-references](#analyzing-cross-references) (`axt`) to locate its usage in the binary. This approach ensures that dynamically resolved or obfuscated functions are not overlooked.
+
+```bash
+f~canEvaluatePolicy
+0x100008360 24 str.canEvaluatePolicy:error:
+```
+
+In the output, you can see a flag `str.canEvaluatePolicy:error:` representing the string `canEvaluatePolicy:error:` which is located at `0x100008360`. You can then use the [`axt` command](https://book.rada.re/analysis/code_analysis.html#analyze-xrefs) to find cross-references to this string using its address:
+
+```bash
+axt @ 0x100008360
+
+(nofunc) 0x100000350 [UNKNOWN] invalid
+sym.func.100004ea8 0x100004f10 [STRING] ldr x1, [x8, 0xa0]
+```
+
+The output shows the cross-references to the given address, indicating where the function is used in the binary in two different locations:
+
+- `0x100000350`: This indicates an invalid reference, possibly due to dynamic resolution.
+- `sym.func.100004ea8`: This indicates a valid reference to a function located at `0x100004ea8`, which can be further analyzed. The reference to the string will be at `0x100004f10`.
+
+### Identifying Strings
+
+**Listing Strings**:
+
+Use the [`iz` command](https://book.rada.re/tools/rabin2/strings.html) to list all strings in the binary. Combine it with [`~` filtering](https://book.rada.re/first_steps/syntax.html#filtering) to filter for specific strings. For example, to search for a string containing `somestring`:
+
+```bash
+iz~+somestring
+```
+
+Alternatively, you can use `izz`
+
+**Searching for Specific Strings**:
+
+Use the [`/` command](https://book.rada.re/search/basic_searches.html) to search for a string directly in the binary:
+
+```bash
+/ somestring
+```
+
+Use advanced search options with [`/`](https://book.rada.re/search/intro.html).
+
+### Analyzing Cross-References
+
+**Finding Cross-References**:
+
+Use the [`axt` command](https://book.rada.re/analysis/code_analysis.html#recursive-analysis) to find cross-references to a specific function. This helps understand how and where the function is invoked.
+
+```bash
+axt @ <function_address>
+```
+
+Note: The `@` symbol is used for [seeking to specific addresses](https://book.rada.re/commandline/seeking.html) in radare2.
```

Comment:
mention that you can also use the "s" command to seek anywhere instead of doing temporal seeks if needed.